### PR TITLE
CLOUP-95681: Split events list command

### PR DESCRIPTION
**Pull request by AndreiMatei2206**
## Proposed changes

_Jira ticket:_ CLOUDP-95681

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by gssbzn — CHANGES_REQUESTED** [review]
It's a bad experience for user who may have scripted the existing events command for it to be removed without warning

Please add a `Deprecated` field to the existing [command](https://pkg.go.dev/github.com/spf13/cobra#Command) and treat `orgs ls` and `projects ls` as new commands

File: internal/cli/events/organization/list.go [hunk 1]
```diff
@@ -87,10 +84,7 @@ func ListBuilder() *cobra.Command {
 		Aliases: []string{"ls"},
 		Args:    require.NoArgs,
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			if opts.orgID != "" && opts.projectID != "" {
-				return fmt.Errorf("both --%s and --%s set", flag.ProjectID, flag.OrgID)
-			}
-			if opts.orgID == "" && opts.projectID == "" {
+			if opts.orgID == "" {
```
Comment (gssbzn) [inline internal/cli/events/organization/list.go:87]: can't highlight all the lines but all of these can now be 
```suggestion
return opts.PreRunE(
				opts.ValidateOrgID,
				opts.initStore,
				opts.InitOutput(cmd.OutOrStdout(), createTemplate),
			)
```

File: internal/cli/events/project/list.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		Use:     "list",
+		Short:   "List events for a project.",
+		Aliases: []string{"ls"},
+		Args:    require.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.projectID == "" {
+				return fmt.Errorf("--%s must be set", flag.ProjectID)
+			}
+			opts.OutWriter = cmd.OutOrStdout()
+			return opts.initStore()
```
Comment (gssbzn) [inline internal/cli/events/project/list.go:92]:
```suggestion
			return opts.PreRunE(
				opts.ValidateProjectID,
				opts.initStore,
				opts.InitOutput(cmd.OutOrStdout(), createTemplate),
			)
```

File: internal/cli/events/events.go [hunk 3]
```diff
@@ -27,7 +29,8 @@ func Builder() *cobra.Command {
 		Short:   "Manage events for your organization or project.",
 	}
 
-	cmd.AddCommand(ListBuilder())
+	cmd.AddCommand(project.Builder())
+	cmd.AddCommand(organization.Builder())
```
Comment (gssbzn) [inline internal/cli/events/events.go:33]:
```suggestion
	cmd.AddCommand(project.Builder(), organization.Builder())
```

File: internal/cli/events/organization/list.go [hunk 4]
```diff
@@ -30,7 +30,6 @@ type ListOpts struct {
 	cli.ListOpts
 	cli.OutputOpts
 	orgID     string
-	projectID string
```
Comment (gssbzn) [inline internal/cli/events/organization/list.go:33]: can't select the line with the problem but `store.EventLister` has probably both methods for projects and orgs, for the new commands you should have a new interface only with the methods you need

File: internal/cli/events/organization/organization.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "organization"
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: cli.GenerateAliases(use),
```
Comment (gssbzn) [inline internal/cli/events/organization/organization.go:26]: we tend to add `orgs` as an alias for organizations
```suggestion
		Aliases: cli.GenerateAliases(use, "orgs"),
```

File: internal/cli/events/project/list.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+)
+
+type ListOpts struct {
+	cli.ListOpts
+	cli.OutputOpts
+	projectID string
+	eventType []string
+	minDate   string
+	maxDate   string
+	store     store.EventLister
```
Comment (gssbzn) [inline internal/cli/events/project/list.go:37]: same as above you want interfaces to be as narrow as possible

File: internal/cli/events/project/list.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type ListOpts struct {
+	cli.ListOpts
+	cli.OutputOpts
+	projectID string
+	eventType []string
+	minDate   string
+	maxDate   string
```
Comment (gssbzn) [inline internal/cli/events/project/list.go:36]: check how `cli.ListOpts` works you could do the same to have a shared struct for the shared things between orgs and projects

File: internal/cli/events/organization/list.go [hunk 8]
```diff
@@ -55,8 +54,6 @@ func (opts *ListOpts) Run() error {
 
 	if opts.orgID != "" {
```
Comment (andreaangiolillo) [inline internal/cli/events/organization/list.go:55]: Once you add `cli.GlobalOpts` to `ListOpts` you should update this piece of code too
```suggestion
	if opts.ValidateOrgID() {
```

Comment (andreaangiolillo) [reply internal/cli/events/organization/list.go:55]: If you implement this comment https://github.com/mongodb/mongocli/pull/780/files#r681786348 you can cancel the `if`

File: internal/cli/events/organization/organization.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 MongoDB Inc
```
Comment (andreaangiolillo) [inline internal/cli/events/organization/organization.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

File: internal/cli/events/project/list.go [hunk 10]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 MongoDB Inc
```
Comment (andreaangiolillo) [inline internal/cli/events/project/list.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

File: internal/cli/events/project/list.go [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type ListOpts struct {
+	cli.ListOpts
+	cli.OutputOpts
+	projectID string
```
Comment (andreaangiolillo) [inline internal/cli/events/project/list.go:33]: same as org
```suggestion
	cli.GlobalOpts
```

File: internal/cli/events/project/list.go [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+{{.ID}}	{{.EventTypeName}}	{{.Created}}{{end}}
+`
+
+func (opts *ListOpts) Run() error {
+	listOpts := opts.newEventListOptions()
+
+	var r *atlas.EventResponse
+	var err error
+
+	if opts.projectID != "" {
```
Comment (andreaangiolillo) [inline internal/cli/events/project/list.go:56]: this `if` can be removed

File: internal/cli/events/project/list_test.go [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+	listOpts.projectID = "1"
+	mockStore.
+		EXPECT().ProjectEvents(listOpts.projectID, listOpts.newEventListOptions()).
+		Return(expected, nil).
+		Times(1)
+
+	if err := listOpts.Run(); err != nil {
+		t.Fatalf("Run() unexpected error: %v", err)
+	}
+}
```
Comment (andreaangiolillo) [inline internal/cli/events/project/list_test.go:47]: we should check the flags used by this command with a `TestListBuilder`. See for an example https://github.com/mongodb/mongocli/blob/6ac1f831b2a7d4ae60f4a7d025e4ab3c8da50d79/internal/cli/atlas/clusters/list_test.go#L51

File: internal/cli/events/project/project.go [hunk 14]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 MongoDB Inc
```
Comment (andreaangiolillo) [inline internal/cli/events/project/project.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

**Review by andreaangiolillo — COMMENTED** [review]
Great work 💯 !! 
I have left only some suggestions! Thanks for doing this PR!

File: e2e/atlas/events_test.go [hunk 15]
```diff
@@ -38,6 +38,7 @@ func TestEvents(t *testing.T) {
 		cmd := exec.Command(cliPath,
 			atlasEntity,
 			eventsEntity,
+			"project",
```
Comment (andreaangiolillo) [inline e2e/atlas/events_test.go:41]: you can use `projectEntity` defined in https://github.com/mongodb/mongocli/blob/6ac1f831b2a7d4ae60f4a7d025e4ab3c8da50d79/e2e/atlas/helper_test.go#L46
```suggestion
			projectEntity,
```

File: e2e/atlas/events_test.go [hunk 16]
```diff
@@ -64,6 +65,7 @@ func TestEvents(t *testing.T) {
 		cmd := exec.Command(cliPath,
 			atlasEntity,
 			eventsEntity,
+			"organization",
```
Comment (andreaangiolillo) [inline e2e/atlas/events_test.go:68]: you can add `organizationEntity` in https://github.com/mongodb/mongocli/blob/6ac1f831b2a7d4ae60f4a7d025e4ab3c8da50d79/e2e/atlas/helper_test.go#L46 as we have for the project and then use it in your test

File: internal/cli/events/organization/list.go [hunk 17]
```diff
@@ -53,11 +47,11 @@ func (opts *ListOpts) Run() error {
 	var r *atlas.EventResponse
 	var err error
 
-	if opts.orgID != "" {
-		r, err = opts.store.OrganizationEvents(opts.orgID, listOpts)
-	} else if opts.projectID != "" {
-		r, err = opts.store.ProjectEvents(opts.projectID, listOpts)
+	if err = opts.ValidateOrgID(); err != nil {
```
Comment (AndreiMatei2206) [inline internal/cli/events/organization/list.go:50]: When I tried addressing the lint error (with the suggestion given by the linter), I got this error:
`shadow: declaration of "err" shadows declaration`

Comment (andreaangiolillo) [reply internal/cli/events/organization/list.go:50]: you have to move 	`var err error` after 
```
	if err := opts.ValidateProjectID(); err != nil {
		return err
	}
```

Comment (andreaangiolillo) [reply internal/cli/events/organization/list.go:50]: same in the org

File: internal/cli/events/organization/list.go [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+	}
+}
+
+// mongocli atlas event(s) organization list [--orgId orgId] [--page N] [--limit N] [--minDate minDate] [--maxDate maxDate].
+func ListBuilder() *cobra.Command {
+	opts := &ListOpts{}
+	opts.Template = listTemplate
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List events for an organization or project.",
```
Comment (andreaangiolillo) [inline internal/cli/events/organization/list.go:80]:
```suggestion
		Short:   "List events for an organization.",
```

File: internal/cli/events/organization/list_test.go [hunk 19]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 MongoDB Inc
```
Comment (andreaangiolillo) [inline internal/cli/events/organization/list_test.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

File: internal/cli/events/organization/organization.go [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "organization"
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: cli.GenerateAliases(use, "orgs"),
+		Short:   "Manage event for your organization.",
```
Comment (andreaangiolillo) [inline internal/cli/events/organization/organization.go:27]:
```suggestion
		Short:   "Manage events for your organization.",
```

File: internal/cli/events/project/list_test.go [hunk 21]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 MongoDB Inc
```
Comment (andreaangiolillo) [inline internal/cli/events/project/list_test.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

**Review by andreaangiolillo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**